Increase upper bound of httparse dependency spec

nox opened the pull request:
Filed in case you don't want #33.

Commit:
Increase upper bound of httparse dependency spec

kdheepak commented:
Can this be merged? I'm getting errors like the following:

```
cargo run
    Updating crates.io index
error: failed to select a version for `httparse`.
    ... required by package `hyperx v1.0.0`
    ... which is depended on by `hubcaps v0.6.0`
    ... which is depended on by `foo v0.1.0 (...)`
versions that meet the requirements `>=1.0, <1.4` are: 1.3.6, 1.3.5, 1.3.4, 1.3.3, 1.3.2, 1.3.1, 1.3.0, 1.2.5, 1.2.4, 1.2.3, 1.2.2, 1.2.1, 1.2.0, 1.1.2, 1.1.1, 1.1.0, 1.0.0

all possible versions conflict with previously selected packages.

  previously selected package `httparse v1.4.0`
    ... which is depended on by `hyper v0.14.9`
    ... which is depended on by `rocket_http v0.5.0-rc.1`
    ... which is depended on by `rocket v0.5.0-rc.1`
    ... which is depended on by `foo v0.1.0 (...)`

failed to select a version for `httparse` which could resolve this conflict
```

Commit:
Increase httparse upper version constraint to <1.6

Commit:
Relax all dependency constraints

Upper bounds suck when they limit semver compatibility, let's
remove them all at once.

dekellum commented:
Partially used dc51351 and released 1.3.1.  httparse 1.5 and any future minor releases will become non-conflicting with a subsequent release including #32.
